inference-engine/src/gna_plugin/frontend/scale_factor_calc.hpp

```diff
@@ -241,49 +486,55 @@ class ScaleFactorPerLayer<InferenceEngine::CNNLayer *> {
         if (cnnLayer->type == "Const") {
             if (quant->_dst_quant.IsScaleSet()) {
                 quant->_src_quant = quant->_dst_quant;
-                return ScaleFactorUpdateResult();
-            }
-
-            auto blob = cnnLayer->blobs["custom"];
-            auto blob_precision = blob->getTensorDesc().getPrecision();
-
-            if (blob_precision != InferenceEngine::Precision::FP32 && blob_precision != InferenceEngine::Precision::FP16) {
-                quant->_dst_quant.SetScale(1.0f);
                 return true;
             }
 
-            if (blob_precision == InferenceEngine::Precision::FP16) {
-                blob = make_fp32_blob(blob);
-            }
-
             auto max_val = std::numeric_limits<float>::min();
             auto min_val = std::numeric_limits<float>::max();
+            if (quant->_dst_quant.IsStatsSet()) {
+                min_val = quant->_dst_quant.GetMinValues().front();
+                max_val = quant->_dst_quant.GetMaxValues().front();
+            } else {
+                auto blob = cnnLayer->blobs["custom"];
+                auto blob_precision = blob->getTensorDesc().getPrecision();
+
+                if (blob_precision != InferenceEngine::Precision::FP32 && blob_precision != InferenceEngine::Precision::FP16) {
+                    quant->_dst_quant.SetScale(1.0f);
+                    return true;
+                }
+
+                if (blob_precision == InferenceEngine::Precision::FP16) {
+                    blob = make_fp32_blob(blob);
+                }
 
-            auto flt_buf = blob->buffer().as<float*>();
-            auto size = blob->size();
+                auto flt_buf = blob->buffer().as<float*>();
+                auto size = blob->size();
 
-            for (int i=0; i < size; i++) {
-                auto val = flt_buf[i];
-                if (val > max_val) max_val = val;
-                if (val < min_val) min_val = val;
+                for (int i = 0; i < size; i++) {
+                    auto val = flt_buf[i];
+                    if (val > max_val) max_val = val;
+                    if (val < min_val) min_val = val;
+                }
             }
 
+            auto levels = fakeQuantize ? MAX_VAL_2B_FEAT : std::numeric_limits<int16_t>::max();
             auto abs_val = std::max(std::abs(max_val), std::abs(min_val));
-            auto scale_val = static_cast<float>(std::numeric_limits<int16_t>::max()) / abs_val;
+            auto scale_val = static_cast<float>(levels) / abs_val;
+            //auto scale_val = (quant->_dst_quant.GetLevels() - 1) / (max_val - min_val);
```

Comment:
Can this commented code be useful for some cases?

Reply:
The commented out formula should be the one that is used for scale factor calculation. However if it is used, it causes degradation of accuracy. So it needs to be investigated.